### PR TITLE
Fix transaction throttler ignoring the initial rate

### PR DESCRIPTION
**Pull request by ejortegau**
## Description
This PR fixes an issue leading to the transaction throttler ignoring the `initial rate` parameter passed to it.

## Related Issue(s)
https://github.com/vitessio/vitess/issues/12549 

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

## Deployment Notes

N/A

### CODE REVIEW

File: go/vt/throttler/throttler.go [hunk 1]
```diff
@@ -130,19 +130,31 @@ func NewThrottler(name, unit string, threadCount int, maxRate, maxReplicationLag
...
+func newThrottlerFromConfig(manager *managerImpl, name, unit string, threadCount int, maxRateModuleMaxRate int64, maxReplicationLagModuleConfig MaxReplicationLagModuleConfig, nowFunc func() time.Time) (*Throttler, error) {
+	err := maxReplicationLagModuleConfig.Verify()
+	if err != nil {
+		return nil, fmt.Errorf("invalid max replication lag config: %w", err)
 	}
-	if maxReplicationLag < 0 {
-		return nil, fmt.Errorf("maxReplicationLag must be >= 0: %v", maxReplicationLag)
+	if maxRateModuleMaxRate < 0 {
+		return nil, fmt.Errorf("maxRate must be >= 0: %v", maxRateModuleMaxRate)
 	}
```
Comment (shlomi-noach) [inline go/vt/throttler/throttler.go:152]: Was this sanity check removed intentionally? I believe we should keep validating that the value is non-negative?

Comment (ejortegau) [reply go/vt/throttler/throttler.go:152]: Yes. this was removed intentionally. Instead, we call `MaxReplicationLagModuleConfig.Verify()` [here](https://github.com/vitessio/vitess/pull/12618/files#diff-8580eaf4909bd8c91212eacb8a23811d625b273a31852a4a18a81c8dde62ca8fR146), which already imposes [stronger constraints](https://github.com/ejortegau/vitess/blob/ejortegau/fix_tx_throttler_ignoring_initial_rate/go/vt/throttler/max_replication_lag_module_config.go#L88-L98) on both target and max replication lag.

File: go/vt/vttablet/tabletserver/txthrottler/tx_throttler_test.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ import (
 	"testing"
 	"time"
 
+	"vitess.io/vitess/go/vt/throttler"
```
[the remark this answers is not in the document]
Comment (ejortegau) [reply go/vt/vttablet/tabletserver/txthrottler/tx_throttler_test.go]: Fixed.

File: go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go [hunk 3]
```diff
@@ -210,8 +210,8 @@ func resetTxThrottlerFactories() {
 	topologyWatcherFactory = func(topoServer *topo.Server, hc discovery.HealthCheck, cell, keyspace, shard string, refreshInterval time.Duration, topoReadConcurrency int) TopologyWatcherInterface {
 		return discovery.NewCellTabletsWatcher(context.Background(), topoServer, hc, discovery.NewFilterByKeyspace([]string{keyspace}), cell, refreshInterval, true, topoReadConcurrency)
 	}
-	throttlerFactory = func(name, unit string, threadCount int, maxRate, maxReplicationLag int64) (ThrottlerInterface, error) {
-		return throttler.NewThrottler(name, unit, threadCount, maxRate, maxReplicationLag)
+	throttlerFactory = func(name, unit string, threadCount int, maxRate int64, maxReplicationLagCondfig throttler.MaxReplicationLagModuleConfig) (ThrottlerInterface, error) {
```
[the remark this answers is not in the document]
Comment (ejortegau) [reply go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go:213]: Fixed.

**Review by harshit-gangal — APPROVED** [review]

**Pull request merged**